Revise targeting docs (#570)
configcat/docs: website/docs/targeting/percentage-options.mdx

---
id: percentage-options
title: Percentage Options
description: Using Percentage Options, you can define that a certain percentage of users should receive a specific value. This way, you can gradually release a new feature to a subset of users.
---

## What are Percentage Options?

Using _Percentage Options_, you can define that a certain percentage of users should receive a specific value.

This way, you can gradually release a new feature to a subset of users. Instead of releasing a feature to all users simultaneously, a specific percentage of users are selected to receive access to the feature. This allows developers to control and monitor the impact of the new feature in a controlled manner.

## How to add Percentage Options?

You can add Percentage Options to a feature flag on the Dashboard by clicking on the **+%** button.

<img alt="Add percentage options" src="/docs/assets/targeting/percentage-options/add-percentage-options_192dpi.png" width="700" height="525" decoding="async" loading="lazy" />

## How does it work? - Anatomy of Percentage Options

> Percentage Options are a list of _% value_ and _served value_ pairs, where % values add up to 100.

Author: ```suggestion
Percentage Options is a list of _% value_ and _served value_ pairs, where % values add up to 100.
```

According to the percentage split defined by the % values, it divides users into groups. The grouping is random and based on the [User Object's](../user-object) **Identifier** attribute by default. You can also use [other user attributes](#percentage-evaluation-attribute) as the basis of the grouping.

If the attribute to use as the basis of the grouping is not present in the [User Object](../user-object), the ["To unidentified" value](../targeting-overview/#to-all-users--to-all-other--to-unidentified-value) will be returned. (Read more about the feature flag evaluation algorithm [here](../feature-flag-evaluation).)

Percentage Options are designed to be [sticky](#stickiness) and [consistent](#consistency) across all SDKs, ensuring a reliable experience.

### % value

Any number between 0 and 100 that represents a randomly allocated group of your users.

### Served value

> The value to return when the user falls into the group determined by the % value.

Author: Koncepcionális probléma a doksiban, hogy ez a "return" nincs kifejtve.

### Number of Percentage Options

> In the case of a feature flag (boolean setting), the number of options must be 2. One for the **ON** and one for the **OFF** state.

Author: ```suggestion
In the case of a feature flag (boolean setting), there must be two options. One for the **ON** and one for the **OFF** state.
```

In the case of a string, integer or double setting, the maximum number of options [depends on your subscription plan](../../subscription-plan-limits). You can add options by clicking the **+% OPTION** button.

## Percentage Options within Targeting Rules

Percentage Options can be used in combination with Targeting Rules.

> In this case, the Percentage Options will be evaluated only when the Targeting Rule matches. In other words, the Percentage Options apply only to the users that matched the Targeting Rule.

Author: ```suggestion
In this case, the Percentage Options will be evaluated only if the Targeting Rule matches. In other words, the Percentage Options apply only to the users that matched the Targeting Rule.
```

If the attribute to use as the basis of the grouping is not present in the [User Object](../user-object), the targeting rule will be skipped - despite the matching Targeting Rule. Read more about the feature flag evaluation algorithm [here](../feature-flag-evaluation).

<img alt="Percentage options within targeting rules" src="/docs/assets/targeting/percentage-options/percentage-options-within-targeting-rule_192dpi.png" width="1084" height="449" decoding="async" loading="lazy" />

## Percentage Evaluation Attribute

The _Percentage Evaluation Attribute_ is the user attribute by which users are grouped. It is the [User Object's](../user-object) **Identifier** attribute by default. You can also use other user attributes as the basis of the grouping (see the example use case [below](#percentage-options-based-on-other-user-attributes)).

### How to change the Percentage Evaluation Attribute?

> Open the kebab menu in the top right corner of the feature flag and choose the **Change percentage attribute** item.

Author: ```suggestion
In the top right corner of the feature flag, open the kebab (3 dots) menu and choose the **Change percentage attribute** item.
```
